test/package-url-resolver_test.js

```diff
@@ -0,0 +1,89 @@
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+const assert = require('chai').assert;
+
+const PackageUrlResolver = require('../lib/package-url-resolver')
+    .PackageUrlResolver;
+
+suite('PackageUrlResolver', function() {
+
+  suite('canResolve', () => {
+
+    test('canResolve is true an in-package URL', () => {
+      let resolver = new PackageUrlResolver();
+      assert.isTrue(resolver.canResolve('foo.html'));
+      assert.isTrue(resolver.canResolve('/foo.html'));
+      assert.isTrue(resolver.canResolve('./foo.html'));
+    });
+
+    test('canResolve is true for a sibling URL', () => {
+      assert.isTrue(new PackageUrlResolver().canResolve('../foo/foo.html'));
+    });
+
+    test('canResolve is false for a cousin URL', () => {
+      assert.isFalse(new PackageUrlResolver().canResolve('../../foo/foo.html'));
+    });
+
+    test('canResolve is false for URL with a hostname', () => {
+      assert.isFalse(new PackageUrlResolver().canResolve('http://abc.xyz/foo.html'));
+    });
+
+    test('canResolve is true for a URL with the right hostname', () => {
+      let resolver = new PackageUrlResolver({
+        hostname: 'abc.xyz',
+      });
+      assert.isTrue(resolver.canResolve('http://abc.xyz/foo.html'));
+      assert.isTrue(resolver.canResolve('http://abc.xyz/./foo.html'));
+      assert.isTrue(resolver.canResolve('http://abc.xyz/../foo.html'));
+      assert.isTrue(resolver.canResolve('http://abc.xyz/foo/../foo.html'));
+    });
+
+  });
+
+  suite('resolve', () => {
+
+    test('resolves an in-package URL', () => {
+      let resolver = new PackageUrlResolver();
+      assert.equal('foo.html', resolver.resolve('foo.html'));
+      assert.equal('foo.html', resolver.resolve('/foo.html'));
+      assert.equal('foo.html', resolver.resolve('./foo.html'));
+    });
+
+    test('resolves a sibling URL', () => {
+      assert.equal(
+          'bower_components/foo/foo.html',
+          new PackageUrlResolver().resolve('../foo/foo.html'));
+    });
+
+    test('throws for a cousin URL', () => {
+      assert.throws(() => new PackageUrlResolver().resolve('../../foo/foo.html'));
+    });
+
+    test('throws for a URL with a hostname', () => {
+      assert.throws(() => new PackageUrlResolver().resolve('http://abc.xyz/foo.html'));
+    });
+
+    test('resolves a URL with the right hostname', () => {
+      let resolver = new PackageUrlResolver({
+        hostname: 'abc.xyz',
+      });
+      assert.equal('foo.html', resolver.resolve('http://abc.xyz/foo.html'));
```

Comment:
No, see the first test: 'resolves an in-package URL'. This test is only for when you explicitly give the resolver a hostname to handle.